ceph/rbd/workflows/rbd.py

```diff
@@ -36,7 +37,179 @@ def create_snap_and_clone(rbd, snap_spec, clone_spec, **kw):
     return 0
 
 
-def config_rbd_multi_pool(
+def create_single_image(
+    config,
+    cluster,
+    rbd,
+    pool,
+    pool_config,
+    image,
+    image_config_val,
+    is_ec_pool,
+    raise_exception=False,
+):
+    """
+    Method to create single image
+    """
+    if not config.get("do_not_create_image"):
+        create_config = {"pool": pool, "image": image}
+        if is_ec_pool:
+            create_config.update({"data-pool": pool_config.get("data_pool")})
+        # Update any other specific arguments that rbd create command takes if passed in config
+        create_config.update(
+            {
+                k: v
+                for k, v in image_config_val.items()
+                if k
+                not in [
+                    "io_total",
+                    "test_config",
+                    "is_secondary",
+                    "snap_schedule_levels",
+                    "snap_schedule_intervals",
+                    "io_size",
+                ]
+            }
+        )
+        create_config.update({"cluster": cluster})
+        out, err = rbd.create(**create_config)
+        if out or err:
+            log.error(f"Image {image} creation failed with err {out} {err}")
+            if raise_exception:
+                raise Exception(f"Image {image} creation failed with err {out} {err}")
+            return 1
+    return 0
+
+
+def create_images(
+    config,
+    cluster,
+    rbd,
+    pool,
+    pool_config,
+    image_config,
+    is_ec_pool,
+    create_image_parallely,
+):
+    """
+    Method to create multiple images parallely/sequentially
+    """
+    if not create_image_parallely:
+        for image, image_config_val in image_config.items():
+            rc = create_single_image(
+                config,
+                cluster,
+                rbd,
+                pool,
+                pool_config,
+                image,
+                image_config_val,
+                is_ec_pool,
+            )
+            if rc:
+                return rc
+    else:
+        try:
+            with parallel() as p:
+                for image, image_config_val in image_config.items():
+                    p.spawn(
+                        create_single_image,
+                        config,
+                        cluster,
+                        rbd,
+                        pool,
+                        pool_config,
+                        image,
+                        image_config_val,
+                        is_ec_pool,
+                        raise_exception=True,
+                    )
+        except Exception:
```

Comment:
does it good to raise exception here instead of return 1

Reply:
You remember in one of your parallel executions as well, we did the same? Because p.spawn does not return any return value? So we have to raise exception?

Reply:
ok got it thanks.